<?xml version="1.0"?>
<robot name="human">

  <link name="base_link">
    <visual>
      <geometry>
        <sphere radius="0.01"/>
      </geometry>
    </visual>
  </link>

  <link name="hip">
    <visual>
      <geometry>  
        <cylinder length="0.204" radius="0.05"/>
     </geometry>
      <origin rpy="1.57075 0 0" xyz="0 0 0" />
    </visual>
  </link>

  <link name="lower_spine"/>
  <joint name="hip_to_lower_spine" type="continuous">
    <parent link="hip"/>
    <child link="lower_spine"/>
  </joint> 

  <link name="middle_spine"/>
  <joint name="lower_spine_to_middle_spine" type="continuous">
    <parent link="lower_spine"/>
    <child link="middle_spine"/>
  </joint> 

  <link name="chest"/>  
  <joint name="middle_spine_to_chest" type="continuous">
    <parent link="middle_spine"/>
    <child link="chest"/>
  </joint> 

  <link name="neck"/>  
  <joint name="chest_to_neck" type="continuous">
    <parent link="chest"/>
    <child link="neck"/>
  </joint> 

 <link name="head"/>  
  <joint name="neck_to_head" type="continuous">
    <parent link="neck"/>
    <child link="head"/>
  </joint> 

 
<!-- -->
 <link name="left_clavicle"/>  
  <joint name="chest_to_left_clavicle" type="continuous">
    <parent link="chest"/>
    <child link="left_clavicle"/>
  </joint> 
  
 <link name="left_shoulder"/>  
  <joint name="left_clavicle_to_left_shoulder" type="continuous">
    <parent link="left_clavicle"/>
    <child link="left_shoulder"/>
  </joint> 
 
 <link name="left_forearm"/>  
  <joint name="left_shoulder_to_left_forearm" type="continuous">
    <parent link="left_shoulder"/>
    <child link="left_forearm"/>
  </joint>

 <link name="left_hand"/>  
  <joint name="left_forearm_to_left_hand" type="continuous">
    <parent link="left_forearm"/>
    <child link="left_hand"/>
  </joint>  
<!-- -->
 <link name="right_clavicle"/>  
  <joint name="chest_to_right_clavicle" type="continuous">
    <parent link="chest"/>
    <child link="right_clavicle"/>
  </joint> 
  
 <link name="right_shoulder"/>  
  <joint name="right_clavicle_to_right_shoulder" type="continuous">
    <parent link="right_clavicle"/>
    <child link="right_shoulder"/>
  </joint> 
 
 <link name="right_forearm"/>  
  <joint name="right_shoulder_to_right_forearm" type="continuous">
    <parent link="right_shoulder"/>
    <child link="right_forearm"/>
  </joint>

 <link name="right_hand"/>  
  <joint name="right_forearm_to_right_hand" type="continuous">
    <parent link="right_forearm"/>
    <child link="right_hand"/>
  </joint>

<!-- -->

<!-- start of right leg geometry-->
  <link name="right_tigh">
    <visual>
      <geometry>
        <cylinder length="0.43" radius="0.05"/>
      </geometry>
    <material name="teal">
      <color rgba="0 .5 .5 1"/>
    </material>
      <origin rpy="0 0 0" xyz="0 0 -0.215"/>    
  </visual>
  </link>

  <link name="right_leg">
    <visual>
      <geometry>
        <cylinder length="0.43" radius="0.05"/>
      </geometry>
      <material name="teal"/>
      <origin rpy="0 0 0" xyz="0 0 -0.215"/>    
   </visual>
  </link>

  <link name="right_foot">
    <visual>
      <geometry>
        <box size="0.15 0.1 0.05"/>
      </geometry>
      <material name="teal"/>
      <origin rpy="0 0 0" xyz="0.075 0 -0.025"/>    
   </visual>
  </link>

 <link name="right_foot_toe"/> 


 <joint name="right_foot_to_right_foot_toe" type="continuous">
    <parent link="right_foot"/>
    <child link="right_foot_toe"/>
    <origin xyz="0 0 -0.45"/>
  </joint> 

 
	

<!-- end of right leg geometry -->
<!-- start of left leg geometry -->
  <link name="left_tigh">
    <visual>
      <geometry>
        <cylinder length="0.43" radius="0.05"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 -0.215"/>    
  </visual>
  </link>

  <link name="left_leg">
    <visual>
      <geometry>
        <cylinder length="0.43" radius="0.05"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 -0.215"/>    
   </visual>
  </link>

  <link name="left_foot">
    <visual>
      <geometry>
        <box size="0.15 0.1 0.05"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0.075 0 -0.025"/>    
   </visual>
  </link>

  <link name="left_foot_toe"/>

  <joint name="left_foot_to_left_foot_toe" type="continuous">
    <parent link="left_foot"/>
    <child link="left_foot_toe"/>
    <origin xyz="0 0 -0.45"/>
  </joint> 

  
  

  <!--link name="left_toe"/>
  <link name="left_toe_end"/-->
    
  
<!-- end of left leg geometry -->

<!-- start of abs -->


  <joint name="base_to_hip" type="continuous">
    <parent link="base_link" />
    <child link="hip" />
    <origin xyz="0 0 0.90"/>
  </joint>
  <!-- right leg joints -->
  
<joint name="hip_to_right_tigh" type="continuous">
	<parent link="hip"/>
    <child link="right_tigh"/>
    <origin xyz="0 -0.1 0"/>
  </joint>





  <joint name="right_tigh_to_right_leg" type="continuous">
    <parent link="right_tigh" />
    <child link="right_leg"/>
    <origin xyz="0 0 -0.45"/>
   </joint>   

  <joint name="right_leg_to_foot" type="continuous">
    <parent link="right_leg"/>
    <child link="right_foot"/>
    <origin xyz="0 0 -0.45"/>
  </joint> 



  <!-- left leg joints -->
   <joint name="hip_to_left_tigh" type="continuous">
    <parent link="hip"/>
    <child link="left_tigh"/>
    <origin xyz="0 0.1 0"/>
  </joint>

  <joint name="left_tigh_to_left_leg" type="continuous">
    <parent link="left_tigh" />
    <child link="left_leg"/>
    <origin xyz="0 0 -0.45"/>
   </joint>   

  <joint name="left_leg_to_foot" type="continuous">
    <parent link="left_leg"/>
    <child link="left_foot"/>
    <origin xyz="0 0 -0.45"/>
  </joint> 

  

 
</robot>

